src/Perlang.Parser/PerlangParser.cs

```diff
@@ -757,5 +742,93 @@ private void Synchronize()
+                // // type identifiers in e.g. `var s: string = "foo"`-style statements.
+                // { "byte", RESERVED_WORD },
+                // { "sbyte", RESERVED_WORD },
+                // { "short", RESERVED_WORD },
+                // { "long", RESERVED_WORD },
+                // { "ushort", RESERVED_WORD },
+                // { "uint", RESERVED_WORD },
+                // { "ulong", RESERVED_WORD },
+                // { "float", RESERVED_WORD },
+                // { "double", RESERVED_WORD },
+                // { "decimal", RESERVED_WORD },
+                // { "char", RESERVED_WORD },
+                //
+                // // Visibility, static/instance, etc
+                // { "public", RESERVED_WORD },
+                // { "private", RESERVED_WORD },
+                // { "protected", RESERVED_WORD },
+                // { "internal", RESERVED_WORD },
+                // { "static", RESERVED_WORD },
+                // { "volatile", RESERVED_WORD },
+                //
+                // // Standard functions
+                // { "printf", RESERVED_WORD },
+                //
+                // // Flow control
+                // { "switch", RESERVED_WORD },
+                // { "break", RESERVED_WORD },
+                // { "continue", RESERVED_WORD },
+                //
+                // // Exception handling
+                // { "try", RESERVED_WORD },
+                // { "catch", RESERVED_WORD },
+                // { "finally", RESERVED_WORD },
+                //
+                // // Asynchronous programming
+                // { "async", RESERVED_WORD },
+                // { "await", RESERVED_WORD },
+                //
+                // // Locking/synchronization
+                // { "lock", RESERVED_WORD },
+                // { "synchronized", RESERVED_WORD },
+                //
+                // // Others
+                // { "mut", RESERVED_WORD },
+                // { "let", RESERVED_WORD },
+                // { "const", RESERVED_WORD },
+                // { "struct", RESERVED_WORD },
+                // { "enum", RESERVED_WORD },
+                // { "sizeof",  RESERVED_WORD },
+                // { "nameof",  RESERVED_WORD },
+                // { "typeof",  RESERVED_WORD },
+                // { "asm", RESERVED_WORD }
+
+                switch (token.Lexeme)
+            {
+                // For details about these, see #178.
+                // case "byte":
+                // case "sbyte":
+                // case "short":
+                case "int":
+                //case "long":
+                //case "ushort":
+                //case "uint":
+                //case "ulong":
+                case "float":
+                //case "double":
+                //case "decimal":
+                //case "char":
+                case "string":
+                    throw Error(token, "Reserved keyword encountered", ParseErrorType.RESERVED_WORD_ENCOUNTERED);
+            }
```

Comment:
...here (continuing from https://github.com/perlang-org/perlang/pull/180/files#r611243898).

Note a few things here though (I'm discovering this as I'm typing): only the type-related reserved words _for existing types_ should go here. Everything else should probably go in the `Scanner` list after all.

Anyway, I think it's best to go the longer route here and add new unit test methods for _each and every reserved keyword_. This has (at least) the following two benefits:

- It ensures that they all work as intended.
- It makes things more fine-grained. If/when we are ready to remove a reserved keyword, either because it's making it into the language proper or because we're scrapping the idea altogether, we'll just remove the test for that keyword in the same MR.

So, this will be cleaned up before I consider this PR ready for being merged.

Reply:
Class-part of this MR extracted to #183 and merged. I'll deal with the remaining unit tests now.